packages/examples/fortune/recording-oracle/src/plugins/curses.ts

```diff
@@ -1,465 +1,14 @@
-  'asshole',
-  'b!tch',
-  'b17ch',
-  'b1tch',
-  'bastard',
-  'bi+ch',
-  'boiolas',
-  'buceta',
-  'c0ck',
-  'cawk',
-  'chink',
-  'cipa',
-  'clits',
-  'cock',
-  'cum',
-  'cunt',
-  'dildo',
-  'dirsa',
-  'ejakulate',
-  'fatass',
-  'fcuk',
-  'fuk',
-  'fux0r',
-  'hoer',
-  'hore',
-  'jism',
-  'kawk',
-  'l3itch',
-  'l3i+ch',
-  'masturbate',
-  'masterbat',
-  'masterbat3',
-  'motherfucker',
-  's.o.b.',
-  'mofo',
-  'nazi',
-  'nigga',
-  'nigger',
-  'nutsack',
-  'phuck',
-  'pimpis',
-  'pusse',
-  'pussy',
-  'scrotum',
-  'sh!t',
-  'shemale',
-  'shi+',
-  'sh!+',
-  'slut',
-  'smut',
-  'teets',
-  'tits',
-  'boobs',
-  'b00bs',
-  'teez',
-  'testical',
-  'testicle',
-  'titt',
-  'w00se',
-  'jackoff',
-  'wank',
-  'whoar',
-  'whore',
-  'damn',
-  'dyke',
-  'fuck',
-  'shit',
-  '@$$',
-  'amcik',
-  'andskota',
-  'arse',
-  'assrammer',
-  'ayir',
-  'bi7ch',
-  'bitch',
-  'bollock',
-  'breasts',
-  'butt-pirate',
-  'cabron',
-  'cazzo',
-  'chraa',
-  'chuj',
-  'Cock',
-  'cunt',
-  'd4mn',
-  'daygo',
-  'dego',
-  'dick',
-  'dike',
-  'dupa',
-  'dziwka',
-  'ejackulate',
-  'Ekrem',
-  'Ekto',
-  'enculer',
-  'faen',
-  'fag',
-  'fanculo',
-  'fanny',
-  'feces',
-  'feg',
-  'Felcher',
-  'ficken',
-  'fitt',
-  'Flikker',
-  'foreskin',
-  'Fotze',
-  'Fu',
-  'fuk',
-  'futkretzn',
-  'gook',
-  'guiena',
-  'h0r',
-  'h4x0r',
-  'hell',
-  'helvete',
-  'hoer',
-  'honkey',
-  'Huevon',
-  'hui',
-  'injun',
-  'jizz',
-  'kanker',
-  'kike',
-  'klootzak',
-  'kraut',
-  'knulle',
-  'kuk',
-  'kuksuger',
-  'Kurac',
-  'kurwa',
-  'kusi',
-  'kyrpa',
-  'lesbo',
-  'mamhoon',
-  'masturbat',
-  'merd',
-  'mibun',
-  'monkleigh',
-  'mouliewop',
-  'muie',
-  'mulkku',
-  'muschi',
-  'nazis',
-  'nepesaurio',
-  'nigger',
-  'orospu',
-  'paska',
-  'perse',
-  'picka',
-  'pierdol',
-  'pillu',
-  'pimmel',
-  'piss',
-  'pizda',
-  'poontsee',
-  'poop',
-  'porn',
-  'p0rn',
-  'pr0n',
-  'preteen',
-  'pula',
-  'pule',
-  'puta',
-  'puto',
-  'qahbeh',
-  'queef',
-  'rautenberg',
-  'schaffer',
-  'scheiss',
-  'schlampe',
-  'schmuck',
-  'screw',
-  'sh!t',
-  'sharmuta',
-  'sharmute',
-  'shipal',
-  'shiz',
-  'skribz',
-  'skurwysyn',
-  'sphencter',
-  'spic',
-  'spierdalaj',
-  'splooge',
-  'suka',
-  'b00b',
-  'testicle',
-  'titt',
-  'twat',
-  'vittu',
-  'wank',
-  'wetback',
-  'wichser',
-  'wop',
-  'yed',
-  'zabourah',
-];
+let pipe =  pipeline('text-classification', 'Rishi-19/Profanity_Test2');
 
 export class Curses {
-  isProfane(text: string) {
-    for (let i = 0; i < curses.length; i++) {
-      const expression = new RegExp(curses[i], 'gi');
-      if (expression.test(text)) {
-        return true;
-      }
-    }
+  async isProfane(text: string) {
+    const result = await (await pipe)(text);
+    const profanity = result[0].find((res: { label: string; }) => res.label === "Profanity_detected");
+    return (profanity && profanity.score > 0.5);
```

Comment:
The score check is not necessary, pipe returns only the most likely label anyways, so score will always be > 0.5, or am I overlooking something?

Reply:
It's alright. In case we want to add different models, I suppose we'd define a common interface for them anyways.